raincloudy/helpers.py

```diff
@@ -29,18 +29,12 @@ def serial_finder(data):
         raise TypeError("Function requires BeautilSoup HTML element.")
 
     try:
-        child = data.find_all('script',
-                              text=re.compile('controller_serial'))[0]
-
-        # pylint: disable=line-too-long
-        regex = re.compile(r"controller_serial':\s*'(?P<controller_serial>\w*)',\s*'faucet_serial':\s*'(?P<faucet_serial>\w*)'") # noqa
-        result = regex.search(child.string).groupdict()
 
         # currently only one faucet is supported on the code
         # we have plans to support it in the future
         parsed_dict = {}
-        parsed_dict['controller_serial'] = result['controller_serial']
-        parsed_dict['faucet_serial'] = [result['faucet_serial']]
+        parsed_dict['controller_serial'] = data.find_all('select', {'name': 'select_controller'})[0].text.strip()
+        parsed_dict['faucet_serial'] = [data.find_all('select', {'name': 'select_faucet'})[0].text.strip()]
```

Comment:
These will pull the currently selected faucet and controller from the UI. It's not super great for people with more than one of either. But I don't think this library currently supports more than one and this unblocks it for now.